### PR TITLE
Add syntax support for env languages

### PR DESCRIPTION
**Pull request by quadratz**
Update: 
We have an even better solution: adding syntax support for `env` languages

Old description:
As a workaround for https://github.com/grammyjs/website/pull/1078, we can simply tell the compiler to treat `env` as plain text.

### CODE REVIEW

**Review by KnorpelSenf — APPROVED** [review]
Sure, that works!

**Comment by KnorpelSenf** [comment]
@quadratz why draft state?

**Comment by quadratz** [comment]
> @quadratz why draft state?

It's ready now. Please review

File: site/docs/.vitepress/config.ts [hunk 1]
```diff
@@ -43,6 +44,8 @@ export default defineConfig({
         aliases: ["asciiart", "ascii", "ftl", "log", "procfile", "text"],
         ...plaintext,
       },
+      // @ts-ignore
```
Comment (KnorpelSenf) [inline site/docs/.vitepress/config.ts:47]: Why does this not compile?

File: site/docs/.vitepress/shared/syntaxes/env.tmLanguage.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
+{
```
Comment (KnorpelSenf) [inline site/docs/.vitepress/shared/syntaxes/env.tmLanguage.json:1]: How did you obtain this definition?

Comment (quadratz) [reply site/docs/.vitepress/shared/syntaxes/env.tmLanguage.json:1]: https://github.com/mikestead/vscode-dotenv/blob/ad506a66ede7d6215cb1f2a16c169eadd414916e/syntaxes/env.YAML-tmLanguage

File: site/docs/.vitepress/config.ts (hunk 1 above)
Comment (quadratz) [reply site/docs/.vitepress/config.ts:47]:
```ansi
Type '{ name: string; scopeName: string; fileTypes: string[]; uuid: string; patterns: ({ comment: string; match: string; name: string; captures: { "1": { name: string; }; }; begin?: undefined; beginCaptures?: undefined; patterns?: undefined; end?: undefined; endCaptures?: undefined; } | { ...; } | { ...; } | { ...; })[]; ...' is not assignable to type 'LanguageInput'.
  Type '{ name: string; scopeName: string; fileTypes: string[]; uuid: string; patterns: ({ comment: string; match: string; name: string; captures: { "1": { name: string; }; }; begin?: undefined; beginCaptures?: undefined; patterns?: undefined; end?: undefined; endCaptures?: undefined; } | { ...; } | { ...; } | { ...; })[]; ...' is not assignable to type 'LanguageRegistration'.
    Types of property 'repository' are incompatible.
      Type '{ interpolation: { comment: string; begin: string; beginCaptures: { "1": { name: string; }; }; patterns: { match: string; captures: { "1": { name: string; }; "2": { name: string; }; }; }[]; end: string; endCaptures: { "1": { name: string; }; }; }; variable: { ...; }; "escape-characters": { ...; }; }' is not assignable to type 'IRawRepository'.
        Type '{ interpolation: { comment: string; begin: string; beginCaptures: { "1": { name: string; }; }; patterns: { match: string; captures: { "1": { name: string; }; "2": { name: string; }; }; }[]; end: string; endCaptures: { "1": { name: string; }; }; }; variable: { ...; }; "escape-characters": { ...; }; }' is missing the following properties from type 'IRawRepositoryMap': $self, $base
```

File: site/docs/.vitepress/shared/syntaxes/env.tmLanguage.json (hunk 2 above)
Comment (KnorpelSenf) [reply site/docs/.vitepress/shared/syntaxes/env.tmLanguage.json:1]: Oh alright

File: site/docs/.vitepress/config.ts (hunk 1 above)
Comment (quadratz) [reply site/docs/.vitepress/config.ts:47]: Sure. Give me a couple of days

Comment (quadratz) [reply site/docs/.vitepress/config.ts]: Their grammar list doesn't seem to strictly follow the type either (see https://github.com/shikijs/textmate-grammars-themes/). To make sure it meets the required properties, I added an empty value to it at https://github.com/grammyjs/website/pull/1083/commits/fbd3565406b0315857c6b865b32f9a58f3c0246e.

**Comment by quadratz** [comment]
https://github.com/grammyjs/website/pull/1083#discussion_r1650079097

**Review by KnorpelSenf — APPROVED** [review]
Awesome stuff, thank you so much!

**Pull request merged**